pkg/xds/sync/egress_proxy_builder.go

```diff
@@ -3,60 +3,144 @@ package sync
 import (
 	"context"
 
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/datasource"
 	"github.com/kumahq/kuma/pkg/core/dns/lookup"
+	"github.com/kumahq/kuma/pkg/core/permissions"
 	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
 	"github.com/kumahq/kuma/pkg/core/resources/manager"
 	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
 	core_store "github.com/kumahq/kuma/pkg/core/resources/store"
 	"github.com/kumahq/kuma/pkg/core/xds"
-	"github.com/kumahq/kuma/pkg/xds/envoy"
+	xds_cache "github.com/kumahq/kuma/pkg/xds/cache/mesh"
+	envoy_common "github.com/kumahq/kuma/pkg/xds/envoy"
+	xds_topology "github.com/kumahq/kuma/pkg/xds/topology"
 )
 
 type EgressProxyBuilder struct {
+	ctx context.Context
+
 	ResManager         manager.ResourceManager
 	ReadOnlyResManager manager.ReadOnlyResourceManager
 	LookupIP           lookup.LookupIPFunc
 	MetadataTracker    DataplaneMetadataTracker
+	meshCache          *xds_cache.Cache
+	DataSourceLoader   datasource.Loader
 
-	apiVersion envoy.APIVersion
-}
-
-func filterMeshesByMTLS(rs core_model.Resource) bool {
-	return rs.(*core_mesh.MeshResource).Spec.GetMtls().GetEnabledBackend() != ""
+	zone       string
+	apiVersion envoy_common.APIVersion
 }
 
-func (p *EgressProxyBuilder) build(key core_model.ResourceKey) (*xds.Proxy, error) {
-	ctx := context.Background()
+func (p *EgressProxyBuilder) Build(key core_model.ResourceKey) (*xds.Proxy, error) {
+	ze := core_mesh.NewZoneEgressResource()
+	if err := p.ReadOnlyResManager.Get(p.ctx, ze, core_store.GetBy(key)); err != nil {
+		return nil, err
+	}
 
-	zoneEgress := core_mesh.NewZoneEgressResource()
-	if err := p.ReadOnlyResManager.Get(ctx, zoneEgress, core_store.GetBy(key)); err != nil {
+	var meshList core_mesh.MeshResourceList
+	if err := p.ReadOnlyResManager.List(
+		p.ctx,
+		&meshList,
+	); err != nil {
```

Comment:
nit: why this and the one below is multiline but 36 is a single line?

Reply:
changed